src/django_rich/test.py

```diff
@@ -0,0 +1,197 @@
+            self.stream.write(capture.get())
+        elif self.dots:
+            with self.console.capture() as capture:
+                self.console.print("x", style=Style(color="yellow"), end="")
+            self.stream.write(capture.get())
+
+    @failfast
+    def addUnexpectedSuccess(self, test: TestCase) -> None:
+        self.unexpectedSuccesses.append(test)
+        if self.showAll:
+            with self.console.capture() as capture:
+                self.console.print("unexpected success", style=Style(color="red"))
+            self.stream.write(capture.get())
+        elif self.dots:
+            with self.console.capture() as capture:
+                self.console.print("u", style=Style(color="red"), end="")
+            self.stream.write(capture.get())
+
+    # Typeshed had wrong signature
+    # https://github.com/python/typeshed/pull/7341
+    def printErrorList(  # type: ignore [override]
+        self,
+        flavour: str,
+        errors: list[tuple[TestCase, str]],
+    ) -> None:
+        for test, err in errors:
+            rule = Rule(style=DJANGO_GREEN)
+            title = f"{flavour}: {self.getDescription(test)}"
+            with self.console.capture() as capture:
+                self.console.print(rule, title, rule)
+            self.stream.write(capture.get())
+            self.stream.write("%s\n" % err)
+
+    def _exc_info_to_string(self, err: _SysExcInfoType, test: TestCase) -> str:
+        """Converts a sys.exc_info()-style tuple of values into a string."""
+        exctype, value, tb = err
+        # Skip test runner traceback levels
+        while tb and self._is_relevant_tb_level(tb):  # type: ignore [attr-defined]
+            tb = tb.tb_next
+
+        msgLines = []
+        if exctype is not None:
+            # assert value is not None
+            # assert tb is not None
+            extract = Traceback.extract(exctype, value, tb, show_locals=True)
+            tb_e = Traceback(extract, suppress=[unittest, testcases])
+            with self.console.capture() as capture:
+                self.console.print(tb_e)
+            msgLines.append(capture.get())
+
+        if self.buffer:
+            assert isinstance(sys.stdout, io.StringIO)
+            output = sys.stdout.getvalue()
+            assert isinstance(sys.stderr, io.StringIO)
+            error = sys.stderr.getvalue()
+            if output:
+                if not output.endswith("\n"):
+                    output += "\n"
+                msgLines.append(STDOUT_LINE % output)
+            if error:
+                if not error.endswith("\n"):
+                    error += "\n"
+                msgLines.append(STDERR_LINE % error)
+
+        return "".join(msgLines)
+
+
+class RichDebugSQLTextTestResult(DebugSQLTextTestResult, RichTextTestResult):
+    def printErrorList(
+        self,
+        flavour: str,
+        errors: list[tuple[TestCase, str], str],
```

Comment:
type hint mistake here: the second `str` is part of the `tuple`, not outside it (fixed)